### ISSUE TITLE
config_stripper no such file in tmp

### ISSUE THREAD

**Issue by Dig-Doug** [bazelbuild/rules_docker#1104]
Hi,

I'm getting an error while trying to build an image:

```bash
ERROR: /tmp/rules_docker_bug/BUILD.bazel:21:1: Action with_deps-with-packages.tar failed (Exit 1) config_stripper failed: error executing command bazel-out/host/bin/external/io_bazel_rules_docker/docker/util/config_stripper '--in_tar_path=bazel-out/k8-fastbuild/bin/with_deps-with-packages.tar.unstripped' ... (remaining 1 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
Traceback (most recent call last):
  File "/home/doug/.cache/bazel/_bazel_doug/93eb6179600f455c74efab96037094a6/sandbox/linux-sandbox/58/execroot/__main__/bazel-out/host/bin/external/io_bazel_rules_docker/docker/util/config_stripper.runfiles/io_bazel_rules_docker/docker/util/config_stripper.py", line 179, in <module>
    sys.exit(main())
  File "/home/doug/.cache/bazel/_bazel_doug/93eb6179600f455c74efab96037094a6/sandbox/linux-sandbox/58/execroot/__main__/bazel-out/host/bin/external/io_bazel_rules_docker/docker/util/config_stripper.runfiles/io_bazel_rules_docker/docker/util/config_stripper.py", line 42, in main
    return strip_tar(args.in_tar_path, args.out_tar_path)
  File "/home/doug/.cache/bazel/_bazel_doug/93eb6179600f455c74efab96037094a6/sandbox/linux-sandbox/58/execroot/__main__/bazel-out/host/bin/external/io_bazel_rules_docker/docker/util/config_stripper.runfiles/io_bazel_rules_docker/docker/util/config_stripper.py", line 61, in strip_tar
    (new_layer_name, new_diff_id) = strip_layer(os.path.join(tempdir, layer))
  File "/home/doug/.cache/bazel/_bazel_doug/93eb6179600f455c74efab96037094a6/sandbox/linux-sandbox/58/execroot/__main__/bazel-out/host/bin/external/io_bazel_rules_docker/docker/util/config_stripper.runfiles/io_bazel_rules_docker/docker/util/config_stripper.py", line 110, in strip_layer
    with tarfile.open(name=path, mode='r') as it:
  File "/usr/lib/python2.7/tarfile.py", line 1675, in open
    return func(name, "r", fileobj, **kwargs)
  File "/usr/lib/python2.7/tarfile.py", line 1740, in gzopen
    fileobj = gzip.GzipFile(name, mode, compresslevel, fileobj)
  File "/usr/lib/python2.7/gzip.py", line 94, in __init__
    fileobj = self.myfileobj = __builtin__.open(filename, mode or 'rb')
IOError: [Errno 2] No such file or directory: u'/tmp/tmpLpEWUP/61af92f6bda87e47c2701156aeb22a663d85b2c13aa19c3f2ecb2c6e4e34a8b8/layer.tar'
----------------
Note: The failure of target @io_bazel_rules_docker//docker/util:config_stripper (with exit code 1) may have been caused by the fact that it is running under Python 2 instead of Python 3. Examine the error to determine if that appears to be the problem. Since this target is built in the host configuration, the only way to change its version is to set --host_force_python=PY3, which affects the entire build.

If this error started occurring in Bazel 0.27 and later, it may be because the Python toolchain now enforces that targets analyzed as PY2 and PY3 run under a Python 2 and Python 3 interpreter, respectively. See https://github.com/bazelbuild/bazel/issues/7899 for more information.
----------------
Target //:with_deps failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2396.172s, Critical Path: 1775.09s
INFO: 57 processes: 57 linux-sandbox.
FAILED: Build did NOT complete successfully
```

I've created a small reproduction [here](https://github.com/Dig-Doug/rules_docker_bug):

```bash
cd /tmp
git clone https://github.com/Dig-Doug/rules_docker_bug.git
cd rules_docker_bug
PULLER_TIMEOUT=1200 bazel build --loading_phase_threads=1 //:with_deps
```

Any idea why this is happening?

Please let me know if you need more information.

Thanks for your help!

**Comment by alex1545** [comment]
Hi @Dig-Doug 
I'm still looking into this issue and haven't figured it out yet.
Are you able to provide an example that installs just a few packages and still produces the same error?

**Comment by Dig-Doug** [comment]
Hi @alex1545,

Thanks for taking a look. I've updated the setup to only install a single package and I still see the issue. One thing I've found is that using a different base image instead of the RBE one does not have the same issue, which makes me think it has something to do with that image.

My reproduction now has 2 targets, `//:works` and `//:fails`, which illustrate using the official Ubuntu image vs the RBE one.

Thanks!
